### PR TITLE
allow haskell-src-exts 1.19

### PR DESCRIPTION
**Pull request by imalsogreg**
Hello! Are you willing to allow this kind of CPP compatibility shimming?

### CODE REVIEW

File: src/Input/Item.hs [hunk 1]
```diff
@@ -178,7 +178,9 @@ hseToSig = tyForall
             TVar a b -> TVar a (b ++ [ty y])
         ty (TyVar _ x) = TVar (fromName x) []
         ty (TyCon _ x) = TCon (fromQName x) []
+#if MIN_VERSION_haskell_src_exts(1,20,0)
         ty (TyInfix an a (UnpromotedName _ b) c) = ty $ let ap = TyApp an in TyCon an b `ap` a `ap` c
+#endif
```
Comment (ndmitchell) [inline src/Input/Item.hs]: Note this probably isn't the right fix - see https://github.com/ndmitchell/hoogle/commit/c4ee561f2dcec959bbd4bc269fb764ece6b7f259#diff-4cbfe74a8c8cd726976adbaf48d37453 for what I had to do to get it into this state.

File: hoogle.cabal [hunk 2]
```diff
@@ -1,7 +1,7 @@
 cabal-version:      >= 1.18
 build-type:         Simple
 name:               hoogle
-version:            5.0.15
+version:            5.0.16
```
Comment (ndmitchell) [inline hoogle.cabal]: As per https://github.com/ndmitchell/neil#contributions - "I only update the .cabal version just before a release" so no need to do it in a PR.

File: src/Input/Item.hs [hunk 3]
```diff
@@ -178,7 +178,9 @@ hseToSig = tyForall
             TVar a b -> TVar a (b ++ [ty y])
         ty (TyVar _ x) = TVar (fromName x) []
         ty (TyCon _ x) = TCon (fromQName x) []
+#if MIN_VERSION_haskell_src_exts(1,20,0)
```
Comment (ndmitchell) [inline src/Input/Item.hs]: See https://github.com/ndmitchell/neil#quirks - "Where I do use CPP, I make sure the code still loads in ghci alone, if all MIN_VERSION_foo macros have not been defined." - so you probably want #if !defined(MIN_VERSION_haskell_src_exts) || ...` so in `ghci` it assumes the newest version.

**Comment by imalsogreg** [comment]
Thanks for catching my mistakes! I should have remembered you keep your `CONTRIBUTING.md` in a central place. I've made the changes you suggested, and tested that the `hseToSig` function I was changing works for the changed constructor (`TyInfix`) under HSE at both 1.91.1 and 1.21.1, in `ghci`. I've rebased my branch with that change.

As for whether this patch helps anyone, it is at the tail end of a yak shave, and I've asked upstream ( https://github.com/NixOS/nixpkgs/pull/32947#issuecomment-353439480 ) whether this is in fact necessary. I don't have any specific need for more flexible HSE bounds except for this one. So I can probably answer the question of whether this patch is useful to us or not sometime Wednesday.

**Comment by ndmitchell** [comment]
Wow, your yak must be mostly bald by now. I'll merge and release later today (just need to find my laptop charger cable first!)

**Comment by ndmitchell** [comment]
Merged, and I made a release of 5.0.16, which I suspect will be required for you to finish off your epic quest. Good luck!

**Comment by imalsogreg** [comment]
Thank you very much! Please feel free to ping me if this causes any maintenance burden down the road, I'd be happy to help relieve annoyance there :)

**Pull request merged**